src/app/reducers/iteration-reducer.ts

```diff
@@ -26,12 +26,15 @@ export const iterationReducer: ActionReducer<IterationState> =
           if (parentIndex > -1) {
             state[parentIndex].hasChildren = true;
           }
-          if (state[parentIndex].children.length) {
-            state[parentIndex].children = [
-              action.payload.iteration,
-              ...state[parentIndex].children
-            ];
-          } else {
+          if(state[parentIndex].children) { 
```

Comment:
Why there is an extra check for the availability of children?

Reply:
There is an extra check for availability of `children` because it throws an undefined error on children when a child iteration is added to a childless iteration as `if(state[parentIndex].children.length >  0 )` will throw an error because a childless iteration doesn't have a property defined called `children`. This property is added in else part of it when the first child is added to a childless iteration. Hence this check `if(state[parentIndex].children` prevents that.

Reply:
There shouldn't have been this check `if(state[parentIndex].children.length) ` but this `if(state[parentIndex].children)`. So I will remove the former.